testing/esp/esdlcmd/baselines/wstest-cpp-installdir/WsTestServiceBase.cpp

```diff
@@ -0,0 +1,527 @@
+            buf.append("</ResultVal>");
+        }
+
+        if (obj->m_ResultArrays)
+        {
+            buf.append("<ResultArrays>");
+            serialize(obj->m_ResultArrays, buf);
+            buf.append("</ResultArrays>");
+        }
+
+        return 0;
+    }
+
+    int serialize(MinVersionReportRequest* obj, StringBuffer& buf)
+    {
+        if (!obj)
+            return 0;
+
+        if (obj->m_RequestString)
+        {
+            buf.append("<RequestString>");
+            buf.append(obj->m_RequestString->str());
+            buf.append("</RequestString>");
+        }
+
+        return 0;
+    }
+
+    int serialize(MinVersionReportResponse* obj, StringBuffer& buf)
+    {
+        if (!obj)
+            return 0;
+
+        if (obj->m_ResponseString)
+        {
+            buf.append("<ResponseString>");
+            buf.append(obj->m_ResponseString->str());
+            buf.append("</ResponseString>");
+        }
+
+        return 0;
+    }
+
+    int serialize(WsTestPingRequest* obj, StringBuffer& buf)
+    {
+        if (!obj)
+            return 0;
+
+        return 0;
+    }
+
+    int serialize(WsTestPingResponse* obj, StringBuffer& buf)
+    {
+        if (!obj)
+            return 0;
+
+        return 0;
+    }
+
+    int serialize(VersionRangeReportRequest* obj, StringBuffer& buf)
+    {
+        if (!obj)
+            return 0;
+
+        if (obj->m_RequestString)
+        {
+            buf.append("<RequestString>");
+            buf.append(obj->m_RequestString->str());
+            buf.append("</RequestString>");
+        }
+
+        return 0;
+    }
+
+    int serialize(VersionRangeReportResponse* obj, StringBuffer& buf)
+    {
+        if (!obj)
+            return 0;
+
+        if (obj->m_ResponseString)
+        {
+            buf.append("<ResponseString>");
+            buf.append(obj->m_ResponseString->str());
+            buf.append("</ResponseString>");
+        }
+
+        return 0;
+    }
+
+};
+int WsTestServiceBase::AllVersionReport(const char* CtxStr, const char* ReqStr, StringBuffer& RespStr)
+{
+    Owned<WsTestUnSerializer> UnSe = new WsTestUnSerializer();
+    Owned<EsdlContext> ctx = new EsdlContext();
+    if (CtxStr && *CtxStr)
+    {
+        Owned<IPropertyTree> ptree = createPTreeFromXMLString(CtxStr);
+        UnSe->unserialize(ctx.get(), ptree.get());
+    }
+    Owned<WsTestServiceBase> service = createWsTestServiceObj();
+    Owned<AllVersionReportRequest> req = new AllVersionReportRequest();
+    Owned<IPropertyTree> ptree;
+    ptree.setown(createPTreeFromXMLString(ReqStr));
+    UnSe->unserialize(req.get(), ptree.get());
+    Owned<AllVersionReportResponse> resp = service->AllVersionReport(ctx.get(), req.get());
+    RespStr = "<Response><Results><Result><Dataset name=\"AllVersionReportResponse\"><Row>";
+    UnSe->serialize(resp.get(), RespStr);
+    RespStr.append("</Row></Dataset></Result></Results></Response>");
+
+    return 0;
+}
```

Comment:
If the service instance (line 453) overloaded a single method name (line 458), this method could be generalized as a template method to eliminate (currently) three additional nearly identical copies.